h5py/tests/test_dataset.py

```diff
@@ -1464,6 +1468,14 @@ def test_astype_wrapper(self):
         arr = dset.astype('f4')[:]
         self.assertArrayEqual(arr, np.arange(100, dtype='f4'))
 
+
+    def test_astype_wrapper_len(self):
+        dset = self.f.create_dataset('x', (100,), dtype='i2')
+        dset[...] = np.arange(100)
+        arr = dset.astype('f4')[:]
+        self.assertEqual(100, len(arr))
```

Comment:
This isn't actually testing the method you added, because you're reading the data to a numpy array first. Try this:

```suggestion
        self.assertEqual(100, len(dset.astype('f4')))
```